File: app/src/main/java/io/lerk/lrkFM/activities/themed/ThemedAppCompatPreferenceActivity.java
package io.lerk.lrkFM.activities.themed;

import android.app.Activity;
import android.content.res.Configuration;
import android.os.Bundle;
import android.preference.PreferenceActivity;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.v7.app.AppCompatDelegate;
import android.view.MenuInflater;
import android.view.View;
import android.view.ViewGroup;

/**
 * A {@link android.preference.PreferenceActivity} which implements and proxies the necessary calls
 * to be used with AppCompat.
 *
 * Now with Themes!
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
public abstract class ThemedAppCompatPreferenceActivity extends PreferenceActivity {

    private AppCompatDelegate delegate;

    /**
     * {@inheritDoc}
     * @see ThemedAppCompatActivity#setThemeFromPreferences(Activity)
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        ThemedAppCompatActivity.setThemeFromPreferences(this);
        getDelegate().installViewFactory();
        getDelegate().onCreate(savedInstanceState);
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onPostCreate(Bundle savedInstanceState) {
        super.onPostCreate(savedInstanceState);
        getDelegate().onPostCreate(savedInstanceState);
    }

    @NonNull
    @Override
    public MenuInflater getMenuInflater() {
        return getDelegate().getMenuInflater();
    }

    @Override
    public void setContentView(@LayoutRes int layoutResID) {
        getDelegate().setContentView(layoutResID);
    }

    @Override
    public void setContentView(View view) {
        getDelegate().setContentView(view);
    }

    @Override
    public void setContentView(View view, ViewGroup.LayoutParams params) {
        getDelegate().setContentView(view, params);
    }

    @Override
    public void addContentView(View view, ViewGroup.LayoutParams params) {
        getDelegate().addContentView(view, params);
    }

    @Override
    protected void onPostResume() {
        super.onPostResume();
        getDelegate().onPostResume();
    }

    @Override
    protected void onTitleChanged(CharSequence title, int color) {
        super.onTitleChanged(title, color);
        getDelegate().setTitle(title);
    }

    @Override
    public void onConfigurationChanged(Configuration newConfig) {
        super.onConfigurationChanged(newConfig);
        getDelegate().onConfigurationChanged(newConfig);
    }

    @Override
    protected void onStop() {
        super.onStop();
        getDelegate().onStop();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        getDelegate().onDestroy();
    }

    public void invalidateOptionsMenu() {
        getDelegate().invalidateOptionsMenu();
    }

    private AppCompatDelegate getDelegate() {
        if (delegate == null) {
            delegate = AppCompatDelegate.create(this, null);
        }
        return delegate;
    }
}

File: app/src/main/java/io/lerk/lrkFM/activities/themed/ThemedAppCompatPreferenceActivity.kt
package io.lerk.lrkFM.activities.themed

import android.content.res.Configuration
import android.os.Bundle
import android.preference.PreferenceActivity
import android.view.MenuInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.appcompat.app.AppCompatDelegate

/**
 * A [android.preference.PreferenceActivity] which implements and proxies the necessary calls
 * to be used with AppCompat.
 *
 * Now with Themes!
 *
 * @author Lukas Fülling (lukas@k40s.net)
 */
abstract class ThemedAppCompatPreferenceActivity : PreferenceActivity() {
    private var delegate: AppCompatDelegate? = null
        private get() {
            if (field == null) {
                field = AppCompatDelegate.create(this, null)
            }
            return field
        }

    /**
     * {@inheritDoc}
     * @see ThemedAppCompatActivity.setThemeFromPreferences
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        ThemedAppCompatActivity.Companion.setThemeFromPreferences(this)
        delegate!!.installViewFactory()
        delegate!!.onCreate(savedInstanceState)
        super.onCreate(savedInstanceState)
    }

    override fun onPostCreate(savedInstanceState: Bundle?) {
        super.onPostCreate(savedInstanceState)
        delegate!!.onPostCreate(savedInstanceState)
    }

    override fun getMenuInflater(): MenuInflater {
        return delegate!!.menuInflater
    }

    override fun setContentView(@LayoutRes layoutResID: Int) {
        delegate!!.setContentView(layoutResID)
    }

    override fun setContentView(view: View) {
        delegate!!.setContentView(view)
    }

    override fun setContentView(view: View, params: ViewGroup.LayoutParams) {
        delegate!!.setContentView(view, params)
    }

    override fun addContentView(view: View, params: ViewGroup.LayoutParams) {
        delegate!!.addContentView(view, params)
    }

    override fun onPostResume() {
        super.onPostResume()
        delegate!!.onPostResume()
    }

    override fun onTitleChanged(title: CharSequence, color: Int) {
        super.onTitleChanged(title, color)
        delegate!!.setTitle(title)
    }

    override fun onConfigurationChanged(newConfig: Configuration) {
        super.onConfigurationChanged(newConfig)
        delegate!!.onConfigurationChanged(newConfig)
    }

    override fun onStop() {
        super.onStop()
        delegate!!.onStop()
    }

    override fun onDestroy() {
        super.onDestroy()
        delegate!!.onDestroy()
    }

    override fun invalidateOptionsMenu() {
        delegate!!.invalidateOptionsMenu()
    }
}
